components/channel/include/channel.h

```diff
@@ -85,14 +87,43 @@ channel_init
 (Channel *ch, const char *identifier, void *ctx, const BaseType_t flags, Channel_callback callback);
 
 /**
- * channel_init_input - helper function to initialize channel input object
+ * channel_init_consumer - helper function to initialize channel object as consumer
  * @ch: pointer to channel object to initialize
  * @identifier: identifier string of the channel
  * @queue: queue handle to send input to 
  */
 static
 void
 inline __attribute__((always_inline))
+channel_init_consumer
+(Channel_consumer *ch, const char *identifier, QueueHandle_t queue)
+{
+    return channel_init(ch, identifier, queue, queueSEND_TO_BACK, &xQueueGenericSend);
+}
+
+/**
+ * channel_init_producer - helper function to initialize channel object as producer
+ * @ch: pointer to channel object to initialize
+ * @identifier: identifier string of the channel
+ */
+static
+void
+inline __attribute__((always_inline)) __attribute__((deprecated))
```

Comment:
why is producer deprecated?
```suggestion
__attribute__((always_inline))
```